### ISSUE TITLE
A global setup hook for each worker

### ISSUE THREAD

**Issue by mitar** [ray-project/ray#2585]
It seems there is no easy way to create a global setup hook which would be run by the worker even before getting any task? This would be useful so that workers are ready and have some data cached in our case so that tasks run much quicker instead of lazy initialization on the first task which we are doing currently.

**Comment by robertnishihara** [comment]
Related to https://github.com/ray-project/ray/issues/144.

**Comment by suquark** [comment]
It could be
```python
import ray

def run_on_all_workers(hook):
    worker = ray.worker.get_global_worker()
    token = ray.utils.random_string()
    worker.redis_client.set(w, 0)

    def for_all_workers(worker_dict):
        world_id = worker_dict['worker'].redis_client.incr(token, 1)  # start-up worker count
        hook(worker_dict['worker'], world_id)

    worker.run_function_on_all_workers(for_all_workers)

def my_hook(worker, world_id):
    print(ray.ObjectID(worker.worker_id), int(world_id))

run_on_all_workers(my_hook)
```

Outputs:
```
ObjectID(f15af35e3e86ea15e577abccf5f02fb4cb8a45e9) 1
ObjectID(06eb6b7e31439d03d51cc8a7c9e6b043c4567a31) 3
ObjectID(5e15fc46cf44a6a5fb6f7cb0e90f90c768d405fc) 2
ObjectID(17b745859410931cea0532ece6e4058367fa22a3) 4
ObjectID(fe12574a91f1411f42b3f0654ecb81a793d9284e) 5
ObjectID(f15af35e3e86ea15e577abccf5f02fb4cb8a45e9) 6
```

**Comment by mitar** [comment]
But if I add more workers later, or workers are restarted because they die, this will not be run, no?

**Comment by suquark** [comment]
This will be run actually. Following previous examples:

```python
import time
@ray.remote
def increase_running_worker():
    time.sleep(5)
    
for _ in range(7):
    increase_worker.remote()
```

outputs:

```
ObjectID(58a826de7b8e6ffff15bffc1ed1f290d5d91652d) 7
ObjectID(7b7a1239933fa5277d2f15c36a4c09d6b4f54b4d) 8
```

But the real problems is it cannot ensure these tasks will be executed before other tasks since they are imported by a thread.

**Comment by mitar** [comment]
I am not sure. You are saying that every time a worker comes around I should just resubmit a task for all workers to run?

**Comment by suquark** [comment]
I means every time a worker comes around, it will run all previous submitted tasks (submitted by run_function_on_all_workers). But we cannot know when it will be run because it is scheduled by a thread of the worker.

**Comment by mitar** [comment]
Oh, I see. I missed `run_function_on_all_workers`. It seems there is already an API for this. Great. I think this will work for us. We can also guard every task just to be sure with initialization.

**Comment by mitar** [comment]
`run_function_on_all_workers` seems to work well. Closing.

**Comment by joyliu329** [comment]
@suquark 
hi,  at the released version: 1.13.0  , worker does't have   the :  ray.worker.get_global_worker()   function . and how can i running a function on all workers now ?
